### PR TITLE
Feature/os utils transfering 

### PR DESCRIPTION
**Pull request by PeepoFrog**
New PR after fixes

### CODE REVIEW

File: pkg/ryokaicommon/utils/os/os_test.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+		}
+	}
+	err = os.RemoveAll(tempDir)
+	if err != nil {
+		t.Error("Unable to delete directory in temporary folder")
+	}
+}
+
+func TestCheckIfPortIsValid(t *testing.T) {
```
Comment (MrLutik) [inline pkg/ryokaicommon/utils/os/os_test.go]: TestCheckIfPortIsValid ->  TestValidatePort

File: pkg/ryokaicommon/utils/os/os_test.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+			}
+
+		}
+	}
+	err = os.RemoveAll(tempDir)
+	if err != nil {
+		t.Error("Unable to delete directory in temporary folder")
+	}
+}
+
```
Comment (MrLutik) [inline pkg/ryokaicommon/utils/os/os_test.go:154]: I would expect also TestValidateIP.

**Review by MrLutik — CHANGES_REQUESTED** [review]

**Review by MrLutik — APPROVED** [review]
Nicely done

File: pkg/ryokaicommon/utils/os/os.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+	return info.IsDir()
+}
+
+// FileExist checks if a file or directory exists at the specified path.
+// It uses os.Stat to attempt to retrieve the file information.
+// If os.Stat returns nil error, the file exists, and the function returns true.
+// If the error is of type os.IsNotExist, indicating the file does not exist, it returns false.
+// For any other type of error, it also returns false, treating inaccessible files as non-existent.
+func FileExist(path string) bool {
```
Comment (andrsj) [inline pkg/ryokaicommon/utils/os/os.go:97]: `PathExists` much better describes the function

File: pkg/ryokaicommon/utils/os/os.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+// The command string is split into the command and its arguments using
+// shlex.Split for proper handling of spaces and quotes.
+// It initializes a new Cmd structure to represent an external command to be executed,
+// passing the command and its arguments separately.
+// If the command execution fails or if there's an error in splitting the command string,
+// it returns an error detailing the issue encountered.
+func RunCommand(command string) ([]byte, error) {
+	args, err := shlex.Split(command)
+	if err != nil {
+		return []byte{}, fmt.Errorf("error when spiting cmd to array of args, err: %w", err)
```
Comment (andrsj) [inline pkg/ryokaicommon/utils/os/os.go:151]: `return nil, fmt.Errorf(...)` looks MUCH better

File: pkg/ryokaicommon/utils/os/os.go [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+// If conversion fails (e.g., input is not an integer), returns false.
+// Checks if the converted integer falls within the valid range for TCP/UDP port numbers (0 to 65535).
+// Returns true if the port number is within the valid range; otherwise, returns false.
+func ValidatePort(input string) bool {
+	port, err := strconv.Atoi(input)
+	if err != nil {
+		return false
+	}
+
+	return !(port < 0 || port > 65535)
```
Comment (andrsj) [inline pkg/ryokaicommon/utils/os/os.go:137]: `return port >= 0 && port <= 65535` easier to read

**Pull request merged**